### ISSUE TITLE
amp-carousel: Best practice: Providing a role and name for carousels

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#29612]
## WCAG Level
Level A

## Priority
Low

## Type
Semantics, Structure

## Pages/Components affected
* [amp-carousel](https://amp.dev/documentation/components/amp-carousel/)

## Description
By default, `<amp-carousel>` is given a `role="list"`, and each item in the carousel a `role="listitem"`. This appropriately conveys the nature of the carousel to assistive technologies. For an `<amp-carousel type="slides">` however, no `role` is included. While visually the carousel is clearly a specific widget/region within the page, structurally the nature of this widget is not conveyed.

## User impact
For users of assistive technologies (such as screen readers) that navigate/read through the page, it will not be obvious when they entered the carousel. Whichever slide is currently shown will seem/appear to be just part of the regular document flow/content. While not a critical problem/failure, this may lead to a confusing experience for these users, who may not be able to discern the distinction between the regular/static page content and the currently displayed slide content.

## Required solution
For slide-based carousels, we recommend adding a `role="region"` to the `<amp-carousel>` element, and giving the region a generic `aria-label="Carousel"`. This way, assistive technologies will explicitly announce when the user navigates into, and out of, the carousel.

Additionally, consider expanding the defined AMP attributes on the element to allow authors to include an explicit name/title for the carousel. If authors supply a name, use this as the `aria-label` for the generated carousel.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied:
1. Turn on your screen reader
2. Load a page with a `<amp-carousel type="slides">`
3. Navigate/read through the page
4. Verify that entering/exiting the carousel leads to an explicit announcement by the screen reader (e.g. in the case of Chrome/NVDA "Carousel region" when reaching the carousel, and "out of region" when exiting the carousel)

## Related standards
* [WCAG 2.1 Success Criterion 4.1.2 Name, Role, Value (Level A)](https://www.w3.org/TR/WCAG21/#name-role-value)

## Test data
Test date: July 2020
